wp-content/plugins/tapestry/endpoints.php

```diff
@@ -6,6 +6,71 @@
 
 require __DIR__ . '/controller/class.tapestry-permissions.php';
 require __DIR__ . '/controller/class.tapestry-controller.php';
+require __DIR__ . '/controller/class.user-controller.php';
+
+ // User endpoints
+
+// Get user progress on a tapestry page by post id. 
+// Example: http://localhost:8888/tapestry-wp/wp-json/tapestry-tool/v1/users/progress?post_id=44
```

Comment:
This is looking much better, thanks! Please just remove `http://localhost:8888/tapestry-wp/` from your comment as the address might be different somewhere else. Don't ever hardcode the website address in the plugin code, even in comments.

Reply:
Removed the http://localhost:8888/tapestry-wp from comments